tests/framework/e2e/e2e.go

```diff
@@ -83,6 +86,7 @@ func (e e2eRunner) NewCluster(ctx context.Context, t testing.TB, opts ...config.
 	default:
 		t.Fatalf("PeerTLS config %q not supported", cfg.PeerTLS)
 	}
+	e2eConfig.BasePort = 11000 + int(atomic.AddInt32(&uniqueCount, int32(cfg.ClusterSize+2)*5)%8000)
```

Comment:
Can you encapsulate the port generation? Something like:
```
uniquePort := uniquePortProvider{
  min: 11000,
  max: 20000,
}

ports := uniquePort.GetBatch(5)
port.Get()
port.Get()
port.Get()
port.Get()
port.Get()
port.Get() <- expect panic
```

Reply:
Batching is not even needed, as there is no requirement of ports being continuous.

Reply:
Example: https://github.com/etcd-io/etcd/blob/main/tests/robustness/identity/id.go

Reply:
Added port allocator that will keep track of free ports.